### PR TITLE
config option to hide the dark overlay & fix #157

### LINKED ISSUE #157: You can enter (and close) rifts in spectator mode

### PR DESCRIPTION
**Pull request by Corail31**
It's mainly to have the possibility to remove this overlay that appears all the time in midnight, without disabling the fancy graphics (and keep it working near world border with the warning distance)

### CODE REVIEW

File: src/main/resources/assets/midnight/lang/en_us.lang [hunk 1]
```diff
@@ -269,17 +269,20 @@ entity.midnight.nova.name=Nova
...
 
 config.midnight.bladeshroom_damage_chance=Bladeshroom Damage Chance
-config.midnight.bladeshroom_damage_chance.tooltip=The chance to take damage when picking a bladeshroom cap
+config.midnight.bladeshroom_damage_chance.tooltip=The chance to take damage when picking a bladeshroom cap.
 
 config.midnight.can_respawn_in_midnight=Allow to respawn in Midnight
 config.midnight.can_respawn_in_midnight.tooltip=If true, players will be allowed to respawn in Midnight.
 
+config.midnight.hide_dark_overlay_in_midnight=Hide the dark overlay in Midnight
+config.midnight.hide_dark_overlay_in_midnight.tooltip=If true, the circular overlay appearing in dark areas will be hidden in Midnight, without the need to disable the fancy graphics.
```
Comment (gegy1000) [inline src/main/resources/assets/midnight/lang/en_us.lang]: Vignette would be a more appropriate description, imo

File: src/main/java/com/mushroom/midnight/common/world/MidnightWorldProvider.java [hunk 2]
```diff
@@ -171,10 +171,13 @@ public boolean shouldClientCheckLighting() {
     @Nullable
     @SideOnly(Side.CLIENT)
     public IRenderHandler getCloudRenderer() {
-        return new IRenderHandler() {
```
Comment (gegy1000) [inline src/main/java/com/mushroom/midnight/common/world/MidnightWorldProvider.java]: Do we need this? Afaik there shouldn't be a cloud renderer if `isSurfaceWorld` is returning false?

File: src/main/java/com/mushroom/midnight/client/ClientEventHandler.java [hunk 3]
```diff
@@ -232,4 +235,17 @@ private static boolean isMusicSound() {
...
         return Arrays.stream(stackTrace).anyMatch(e -> e.getClassName().equals(MusicTicker.class.getName()));
     }
+
+    @SubscribeEvent(priority=EventPriority.LOWEST)
+    public static void onRenderVignetteOverLay(RenderGameOverlayEvent.Pre event) {
+        if (MidnightConfig.hideDarkOverlayInMidnight && event.getType() == RenderGameOverlayEvent.ElementType.VIGNETTE) {
+            if (Helper.isMidnightDimension(MC.world)) {
+                WorldBorder worldborder = MC.world.getWorldBorder();
+                float distWarn = Math.max(worldborder.getWarningDistance(), (float) Math.min(worldborder.getResizeSpeed() * worldborder.getWarningTime() * 1000d, Math.abs(worldborder.getTargetSize() - worldborder.getDiameter())));
+                if (worldborder.getClosestDistance(MC.player) >= distWarn) {
```
Comment (gegy1000) [inline src/main/java/com/mushroom/midnight/client/ClientEventHandler.java]: Hmm would this not cause the effect to suddenly appear when near a worldborder?

File: src/main/java/com/mushroom/midnight/common/world/MidnightWorldProvider.java (hunk 2 above)
Comment (Corail31) [reply src/main/java/com/mushroom/midnight/common/world/MidnightWorldProvider.java]: yes but it's called first any way before isSurface

File: src/main/java/com/mushroom/midnight/client/ClientEventHandler.java (hunk 3 above)
Comment (Corail31) [reply src/main/java/com/mushroom/midnight/client/ClientEventHandler.java]: test

Comment (Corail31) [reply src/main/java/com/mushroom/midnight/client/ClientEventHandler.java]: image

**Pull request merged**